core/arch/arm/tee/entry_std.c

```diff
@@ -169,6 +168,10 @@ static TEE_Result copy_in_params(const struct optee_msg_param *params,
 		}
 	}
 
+	for (; n < TEE_NUM_PARAMS; n++) {
```

Comment:
I agree that this needs to be fixed and that your proposal takes care of the issue, but what about just simply do this at the beginning of the function:
```bash
uint8_t pt[TEE_NUM_PARAMS] = { 0 };
```

Reply:
This was the initial method I considered -- the reasoning I went away from it was that it makes an unstated assumption that `TEE_PARAM_TYPE_NONE` will always be 0.

However, I'm completely open to changing it, completely up to the codestyle preferences here.

Reply:
`TEE_PARAM_TYPE_NONE` is mandated to be zero by the TEE Internal Core API spec, so `= { 0 };` is safe.